src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/Assertions.java

```diff
@@ -65,6 +71,57 @@ public void taskStateWasUpdatedInDatabase(String taskId, String value, Headers a
             .log();
     }
 
+    public void taskValuesWasUpdatedInDatabase(String taskId, Map<String, Matcher<?>> fieldValueMap,
+                                               Headers authenticationHeaders) {
+
+        Response result = restApiActions.get(
+            "task/{task-id}",
+            taskId,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .and().contentType(MediaType.APPLICATION_JSON_VALUE).log();
+
+        fieldValueMap.entrySet().forEach(
+            entry -> result.then().assertThat()
+                .body(entry.getKey(), entry.getValue()).log()
+        );
+    }
+
+    public void taskActionAttributesUpdatedInDatabase(String taskId, String assigneeId,
+                                                      String taskState, String lastUpdatedUserId, TaskAction taskAction,
+                                                      TestAuthenticationCredentials taskRetrieveCredentials) {
+
+        Map<String, Matcher<?>> taskValueMap = Map.of(
+            "task.id", equalTo(taskId),
+            "task.task_state", CoreMatchers.is(taskState),
+            "task.assignee", equalTo(assigneeId),
+            "task.last_updated_timestamp", notNullValue(),
+            "task.last_updated_user", equalTo(lastUpdatedUserId),
+            "task.last_updated_action", equalTo(taskAction.getValue())
+        );
+
+        taskValuesWasUpdatedInDatabase(taskId, taskValueMap, taskRetrieveCredentials.getHeaders());
+    }
+
+    public void nullTaskActionAttributesUpdatedInDatabase(String taskId, String taskState,
+                                                          TestAuthenticationCredentials taskRetrieveCredentials) {
+
+        Map<String, Matcher<?>> taskValueMap = Map.of(
+            "task.id", equalTo(taskId),
+            "task.task_state", CoreMatchers.is(taskState),
+            "task.assignee", nullValue(),
+            "task.last_updated_timestamp", nullValue(),
+            "task.last_updated_user", nullValue(),
+            "task.last_updated_action", nullValue()
```

Comment:
Just to say when we switch this on we don't want the last_updated values to ever be `null` in the DB.  This makes things a lot simpler for CTSC MI.

Reply:
true, as we are currently doing one endpoint at a time, after we address all the endpoints probably this is true that these attributes wont be null, but at this point of time when task is initially created these values are null hence asserting it as null, once this method is no longer used will remove.

Reply:
this will be addressed in the configuration and auto-assignment story as discussed today